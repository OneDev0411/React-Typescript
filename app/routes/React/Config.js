// Config.js
import React from 'react'
import { Route, IndexRoute, Redirect } from 'react-router'

// Main component
import App from '../../components/App'

// Pages
import Landing from '../../components/Pages/Landing'
import Recents from '../../components/Pages/Dashboard/Recents'
import Mls from '../../components/Pages/Dashboard/Mls'
import Contacts from '../../components/Pages/Dashboard/Contacts'
import Tasks from '../../components/Pages/Dashboard/Tasks'
import Transactions from '../../components/Pages/Dashboard/Transactions'
import SingleTransaction from '../../components/Pages/Dashboard/Transactions/Single'
import NewTransaction from '../../components/Pages/Dashboard/Transactions/New'
import SignUp from '../../components/Pages/SignUp'
import SignIn from '../../components/Pages/SignIn'
import Verify from '../../components/Pages/Verify'
import Password from '../../components/Pages/Password'
import NoMatch from '../../components/Pages/NoMatch'
import Settings from '../../components/Pages/Account/Settings'
import Notifications from '../../components/Pages/Account/Notifications'
import Recommend from '../../components/Pages/Dashboard/Mls/Listing/Recommend'
import Agents from '../../components/Pages/Dashboard/Mls/Agents'

export default (
  <Route path="/" component={App}>
    <IndexRoute component={Landing}/>
    <Route path="signup" component={SignUp}/>
    <Route path="signin" component={SignIn}/>
    <Route path="account/settings" component={Settings}/>
    <Route path="account/notifications" component={Notifications}/>
    <Redirect from="dashboard" to="dashboard/recents" />
    <Route path="dashboard/recents" component={Recents}/>
    <Route path="dashboard/recents/:room_id" component={Recents}/>
    <Route path="dashboard/mls" component={Mls}/>
    <Route path="dashboard/mls/agents" component={Agents}/>
    <Route path="dashboard/mls/listing/recommend" component={Recommend}/>
    <Route path="dashboard/contacts" component={Contacts}/>
    <Route path="dashboard/contacts/:id" component={Contacts}/>
    <Route path="dashboard/tasks" component={Tasks}/>
    <Route path="dashboard/transactions" component={Transactions}/>
    <Route path="dashboard/transactions/new" component={NewTransaction}/>
    <Route path="dashboard/transactions/:id" component={SingleTransaction}/>
    <Route path="dashboard/transactions/:id/attachments/:id" component={SingleTransaction}/>
    <Route path="verify/:slug" component={Verify}/>
    <Route path="password/:slug" component={Password}/>
    <Route path="*" component={NoMatch}/>
  </Route>
)