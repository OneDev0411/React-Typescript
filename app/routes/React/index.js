// index.js
import React from 'react'
import { match, RoutingContext } from 'react-router'
import ReactDOMServer from 'react-dom/server'

// Store
import AppStore from '../../stores/AppStore'

// Config
import routes from './Config'

module.exports = (app, config) => {

  app.get('*',(req, res) => {

    match({ routes, location: req.url }, (error, redirectLocation, renderProps) => {
      
      let reactMarkup = ReactDOMServer.renderToStaticMarkup(<RoutingContext data={AppStore.data} {...renderProps} />)
      
      res.locals.reactMarkup = reactMarkup

      console.log(reactMarkup)

      if (error) {
      
        res.status(500).send(error.message)
      
      } else if (redirectLocation) {
        
        res.redirect(302, redirectLocation.pathname + redirectLocation.search)

      } else if (renderProps) {
        
        // Success!
        res.status(200).render('index.html')
      
      } else {

        res.status(404).render('index.html')
      
      }

    })
  })
  
}