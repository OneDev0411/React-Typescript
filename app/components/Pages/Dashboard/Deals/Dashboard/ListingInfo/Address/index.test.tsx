import React from 'react'
import { render } from 'enzyme'
import toJson from 'enzyme-to-json'

import withAddress from 'fixtures/deal/live-seller'
import withoutAddress from 'fixtures/deal/live-seller-without-address'

import { Address } from '.'

describe('Deal address component', () => {
  it('renders with address', () => {
    const wrapper = render(<Address deal={withAddress} />)

    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders without address', () => {
    const wrapper = render(<Address deal={withoutAddress} />)

    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders button with "Edit Address" CTA for deals with address', () => {
    const wrapper = render(<Address deal={withAddress} />)

    expect(
      wrapper
        .children()
        .last()
        .text()
    ).toBe('Edit Address')
  })

  it('renders button with "+ Add Address" CTA for deals without address', () => {
    const wrapper = render(<Address deal={withoutAddress} />)

    expect(
      wrapper
        .children()
        .last()
        .text()
    ).toBe('+ Add Address')
  })
})
