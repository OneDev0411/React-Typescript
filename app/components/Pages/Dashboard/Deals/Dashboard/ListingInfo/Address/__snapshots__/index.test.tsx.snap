// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deal address component renders with address 1`] = `
<div
  class="makeStyles-container-1"
  title=""
>
  <h6
    class="MuiTypography-root-15 MuiTypography-h6-25"
  >
    1200 Main St
  </h6>
  <button
    class="MuiButtonBase-root-74 MuiButton-root-45 MuiButton-text-47 makeStyles-editButton-2 MuiButton-textSecondary-49 MuiButton-textSizeSmall-60 MuiButton-sizeSmall-66"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label-46"
    >
      Edit Address
    </span>
  </button>
</div>
`;

exports[`Deal address component renders without address 1`] = `
<div
  class="makeStyles-container-77"
  title=""
>
  <h6
    class="MuiTypography-root-91 MuiTypography-h6-101"
  >
    Some title for a deal without address
  </h6>
  <button
    class="MuiButtonBase-root-150 MuiButton-root-121 MuiButton-text-123 makeStyles-editButton-78 MuiButton-textSecondary-125 MuiButton-textSizeSmall-136 MuiButton-sizeSmall-142"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label-122"
    >
      + Add Address
    </span>
  </button>
</div>
`;
